error[E0573]: expected type, found variant `Err`

  --> mutant.rs:50:43

   |

50 | fn into_future ( self ) -> FutureResult < Err , E > {

   |                                           ^^^ not a type

   |

thread 'rustc' panicked at 'assertion failed: !self.substitutions.is_empty()', compiler/rustc_errors/src/lib.rs:173:9

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace



error: internal compiler error: unexpected panic



note: the compiler unexpectedly panicked. this is a bug.



note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md



note: rustc 1.49.0-nightly (4f7612ac1 2020-10-31) running on x86_64-unknown-linux-gnu



note: compiler flags: -C debuginfo=2 -C opt-level=1 -C target-cpu=skylake --crate-type staticlib



query stack during panic:

end of query stack

